# Librerias

In [1]:
! pip install pyarrow


[notice] A new release of pip is available: 26.0.1 -> 26.1
[notice] To update, run: python.exe -m pip install --upgrade pip


In [2]:
import pandas as pd

# Carga de la bbdd

In [3]:
df=pd.read_parquet("../../data/df_limpiado_260427.parquet")
df = df.reset_index()
df

,index,age,job,marital,education,default,balance,housing,loan,contact,day,month,duration,campaign,pdays,previous,poutcome,deposit,contactado_campania_previa
0,0,59,administration,married,secondary,no,2343,yes,no,unknown,5,may,1042,1,NaN,0,sin_campania_previa,yes,0
1,1,56,administration,married,secondary,no,45,no,no,unknown,5,may,1467,1,NaN,0,sin_campania_previa,yes,0
2,2,41,technician,married,secondary,no,1270,yes,no,unknown,5,may,1389,1,NaN,0,sin_campania_previa,yes,0
3,3,55,services,married,secondary,no,2476,yes,no,unknown,5,may,579,1,NaN,0,sin_campania_previa,yes,0
4,4,54,administration,married,tertiary,no,184,no,no,unknown,5,may,673,2,NaN,0,sin_campania_previa,yes,0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
10982,10982,40,management,married,secondary,no,8486,no,no,unknown,6,may,260,3,NaN,0,sin_campania_previa,no,0
10983,10983,53,management,married,tertiary,no,20772,no,no,cellular,4,feb,715,1,NaN,0,sin_campania_previa,no,0
10984,10984,55,blue-collar,married,primary,no,3297,yes,yes,telephone,30,apr,96,1,NaN,0,sin_campania_previa,no,0
10985,10985,41,management,married,tertiary,no,9,yes,no,cellular,22,jul,82,3,NaN,0,sin_campania_previa,no,0


# KPI's:

## 1. Porcentaje de conversión a depósito

In [4]:
# Número de clientes que suscribieron
num_suscriptores=(df['deposit'] == 'yes').sum()
print("El número de clientes que suscribieron es: ",num_suscriptores)

#Número total de clientes contactados
clientes_contactados=len(df)
print("El número total de clientes contactados es: ", clientes_contactados)

#Porcentaje de conversión a depósito (kpi1)
kpi1 = round(num_suscriptores / clientes_contactados * 100, 2)
print("KPI 1: El porcentaje de conversión a depósito es ",kpi1,"%")


El número de clientes que suscribieron es:  5289
El número total de clientes contactados es:  10987
KPI 1: El porcentaje de conversión a depósito es  48.14 %


## 2. Promedio de duración de llamadas de suscriptores

In [5]:
#Sumatoria de la duración de llamadas de suscriptores
duracion_total_llam=df[df['deposit'] == 'yes']['duration'].sum()
print("La duración total de las llamadas a suscriptores es:",duracion_total_llam)

#Número de clientes que suscribieron
clientes_suscritos=(df['deposit'] == 'yes').sum() 

kpi2=round(duracion_total_llam/clientes_suscritos,1)
print(f"KPI 2: Duración media llamadas suscriptores: {kpi2} segundos ({round(kpi2/60,1)} min)")

La duración total de las llamadas a suscriptores es: 2801017
KPI 2: Duración media llamadas suscriptores: 529.6 segundos (8.8 min)


## 3. Porcentaje de llamadas a teléfono o móvil

In [6]:
count_tlf = df[df['contact'] == 'telephone']['campaign'].sum()
count_mvl = df[df['contact'] == 'cellular']['campaign'].sum() 
count_llamadas = df['campaign'].sum()

porcentaje_tlf = (count_tlf / count_llamadas)*100
porcentaje_mvl = (count_mvl / count_llamadas)*100

kpi3_tlf = porcentaje_tlf
kpi3_mvl = porcentaje_mvl

print(f"El porcentaje de llamadas a teléfono es del {round(kpi3_tlf,2)}%.")
print(f"El porcentaje de llamadas a móvil es del {round(kpi3_mvl,2)}%.")

El porcentaje de llamadas a teléfono es del 8.15%.
El porcentaje de llamadas a móvil es del 69.42%.


## 4. Promedio de contactos previos de suscriptores

In [7]:
contactos_previos = df[df['deposit'] == 'yes']['previous'].sum()

promedio_contactos = (contactos_previos / num_suscriptores)

kpi4 = promedio_contactos

print(f"El promedio de contactos previos es de {round(kpi4,2)}.")

El promedio de contactos previos es de 1.11.


## 5. Mes con mayor tasa de conversión

In [8]:
suscriptores_mes = df[df['deposit'] == 'yes'].groupby('month', observed=True).size()
contactados_mes = df.groupby('month', observed=True).size()
conversion_mes = suscriptores_mes / contactados_mes

max_mes = conversion_mes.idxmax()
max_conversion = (conversion_mes.max())*100

kpi5_mes = str(max_mes).upper()
kpi5_tasa = max_conversion

print(f"El mes con mayor tasa de conversión es {kpi5_mes} con un {round(kpi5_tasa,2)}%.")

El mes con mayor tasa de conversión es DEC con un 90.91%.


# Conclusiones

La campaña muestra una tasa de conversión alta (48.1%). El canal móvil concentra el 69.4% de los contactos mientras que el porcentaje de llamadas a teléfono es del 8.2%. Los suscriptores tienen conversaciones largas (8.8 min) y un historial previo con el banco (1.11 contactos). Diciembre es el mes de mayor conversión con un 90.9%."